### PR TITLE
📦 Update @metamask/detect-provider to 2.0.0

### PR DESCRIPTION
**Pull request by gsteenkamp89**
# Bump @metamask/detect-provider version to 2.0.0

This will resolve [this](https://github.com/TrueFiEng/useDApp/issues/529) issue relating to sourcemaps.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "usedapp-generate-hooks": "./bin/usedapp-generate-hooks.js"
   },
   "dependencies": {
-    "@metamask/detect-provider": "^1.2.0",
+    "@metamask/detect-provider": "2.0.0",
```
Comment (yivlad) [inline packages/core/package.json:20]: Is there a reason why the version is fixed?

**Review by yivlad — CHANGES_REQUESTED** [review]
Please also add changeset

File: packages/core/package.json (hunk 1 above)
Comment (gsteenkamp89) [reply packages/core/package.json:20]: shouldn't be, I'll add the caret back! 👍

**Review by yivlad — COMMENTED** [review]
@gsteenkamp89 please also update the lockfile - CI seems to fail because of this

**Comment by gsteenkamp89** [comment]
> @gsteenkamp89 please also update the lockfile - CI seems to fail because of this

@yivlad done 👍

**Review by yivlad — APPROVED** [review]

**Comment by tamebadger** [comment]
what's the timeline on a release that includes this fix out of interest sake @yivlad ?

**Comment by yivlad** [comment]
Hey @tamebadger, we'll publish the latest version to npm when we'll be sure that the version is stable. For now you're always welcome to download `dev` version from npm - it's the latest master from this repo.

**Pull request merged**